### PR TITLE
SDIT-1351 Activities migration - find activities with missing pay bands

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivitiesResource.kt [hunk 1]
```diff
@@ -477,6 +478,48 @@ class ActivitiesResource(
   ): List<FindSuspendedAllocationsResponse> =
     allocationService.findSuspendedAllocations(prisonId, excludeProgramCodes, courseActivityId)
 
+  @PreAuthorize("hasRole('ROLE_NOMIS_ACTIVITIES')")
+  @GetMapping("/allocations/missing-pay-bands")
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivitiesResource.kt:482]: The main change is to add a new endpoint to be called from the activities migration preview and provide a list of allocated prisoners who don't have a valid pay band (but there is a pay rate available on the activity, e.g. not interested in Unemployed which doesn't pay anyway).

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/GetAllocationResourceIntTest.kt [hunk 2]
```diff
@@ -241,6 +244,12 @@ class GetAllocationResourceIntTest : IntegrationTestBase() {
               courseAllocation(courseActivity = courseActivity, startDate = "$today", suspended = true)
             }
           }
+          offender {
+            booking {
+              incentive(iepLevelCode = "BAS")
+              courseAllocation(courseActivity = courseActivity, startDate = "$today")
+            }
+          }
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/GetAllocationResourceIntTest.kt:252]: Add a missing pay band allocation test case to each of the allocation selection tests to make sure that we only include allocations that are in the migration.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/GetAllocationResourceIntTest.kt [hunk 3]
```diff
@@ -817,6 +974,266 @@ class GetAllocationResourceIntTest : IntegrationTestBase() {
       .exchange()
       .expectStatus().isOk
 
+  /*
+   * This endpoint is closely related to `GET /allocations/ids` and any changes to the allocation selection rules should
+   * affect it. For this reason I've added an allocation with a missing pay band to each of the tests in the `AllocationSelection`
+   * nested class above to further test this endpoint.
+   */
+  @Nested
+  @DisplayName("GET /allocations/missing-pay-bands")
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/GetAllocationResourceIntTest.kt:983]: Tests the various edge cases where a specific allocation does or doesn't have a missing pay band.

**Review by petergphillips — APPROVED** [review]
LGTM
